### DISCUSSION TITLE
Segmention Fault P.2

### DISCUSSION THREAD

**Discussion by SantiagoOsorio3** [IIC2133-PUC/2024-2#85] [Tarea 1] [answered]
Hola, con respecto a la respuesta que me dieron en la discussion 80, volvi a correr valgrind para encontrar el segmention fault pero sigo sin poder encontrar el error. Nose si es que podria ser por otra cosa (adjunto aqui lo que me arroja valgrind). 
PD: el segmention fault es en vscode...

valgrind --show-leak-kinds=all ./program input.txt output.txt
==94== Memcheck, a memory error detector
==94== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==94== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==94== Command: ./program input.txt output.txt
==94==
ENTER
ENTER
ENTER
ENTER
ENTER
LEAVE
FESTIVAL
HAPPINESS-BUFF
1
==94==
==94== HEAP SUMMARY:
==94== in use at exit: 0 bytes in 0 blocks
==94== total heap usage: 14 allocs, 14 frees, 6,392 bytes allocated
==94==
==94== All heap blocks were freed -- no leaks are possible
==94==
==94== For lists of detected and suppressed errors, rerun with: -s
==94== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
root@9806622cfb78:~/volume/probar_setup#

**Comment by tomastrivino** [comment, marked as answer]
Hola, según tu output
```
==94== All heap blocks were freed -- no leaks are possible
==94==
==94== For lists of detected and suppressed errors, rerun with: -s
==94== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```
No hay errores presentes (al menos al ejecutarlo como ` ./program input.txt output.txt`, ojo, puede ser que ` ./program input1.txt output.txt` no tenga segfault pero ` ./program input2.txt output.txt` sí).

Si al ejecutar tu programa te ocurre un Segmentation Fault, entonces asegúrate de ejecutar Valgrind usando ese mismo comando, por la razón que expuse en el párrafo de arriba.

Saludos!

**Reply by tomastrivino** [reply]
Lo otro, ¿ a qué te refieres con segmentation fault en vscode?

**Reply by SantiagoOsorio3** [reply]
me refiero que el codigo al correrlo dentro del vscode como ./happy input.txt output.txt me sale el segmention fault, pero al correrlo en edd-docker todo bien...
